src/java/org/apache/cassandra/db/commitlog/CommitLogSegmentManagerCDC.java

```diff
@@ -78,11 +81,45 @@ public void discard(CommitLogSegment segment, boolean delete)
                 FileUtils.deleteWithConfirm(cdcLink);
 
             File cdcIndexFile = segment.getCDCIndexFile();
-            if (cdcIndexFile.exists())
-                FileUtils.deleteWithConfirm(cdcIndexFile);
+            deleteCDCFiles(cdcLink, cdcIndexFile);
         }
     }
 
+    /**
+     * Delete the oldest CDC commit log segment to free up space.
+     * @return total deleted file size in bytes
+     */
+    public long deleteOldestCommitLogSegment()
+    {
+        File cdcDir = new File(DatabaseDescriptor.getCDCLogLocation());
+        Preconditions.checkState(cdcDir.isDirectory(), "The CDC directory does not exist.");
+        File[] files = cdcDir.tryList(f -> CommitLogDescriptor.isValid(f.name()));
+        Preconditions.checkState(files != null && files.length > 0, "There should be at least 1 CDC commit log segment.");
+        List<File> sorted = Arrays.stream(files)
+                                  .sorted(Comparator.comparingLong(File::lastModified))
+                                  .collect(Collectors.toList());
+        File oldestCdcFile = sorted.get(0);
+        File cdcIndexFile = CommitLogDescriptor.inferCdcIndexFile(oldestCdcFile);
+        return deleteCDCFiles(oldestCdcFile, cdcIndexFile);
+    }
+
+    private long deleteCDCFiles(File cdcFile, File cdcIndexFile)
+    {
+        long total = 0;
+        if (cdcFile.exists())
+        {
+            total += cdcFile.length();
+            FileUtils.deleteWithConfirm(cdcFile);
```

Comment:
Looks like this is deprecated as of CEP-10. Is there another call we should use here? (a cursory inspection didn't surface anything for me, but the method is definitely flagged as Deprecated...)

Reply:
👍 Now that there is a `File` wrapper class, `org.apache.cassandra.io.util.File`, instead of `java.io.File`. I think the intent of the deprecation is to call `delete` on the file wrapper directly.